Tests/SemanticTests/invalid_types.flint

```diff
@@ -20,6 +22,12 @@ InvalidTypes :: (any) {
     var a: Bool = true
     a = b // expected-error {{Use of undeclared identifier 'b'}}
 
+    // expected-error@27 {{Incompatible assignment between values of type 'Address' and 'Bool'}}
+    // expected-error@27 {{Cannot convert expression of type 'Bool' to expected subscript type 'Int'}}
+    arr[a] = a
+
+    dict[a] // expected-error {{Cannot convert expression of type 'Bool' to expected subscript type 'Address'}}
```

Comment:
Nice. Could you add a line to check there aren't implicit conversions between `Address` and `Int`? Like `arr[3] = ...`

Reply:
arr[3] = ... is valid, do you mean dict[3]?